src/sagas.js

```diff
@@ -286,27 +326,66 @@ function* fetchVmsByCountVLower (action) {
       yield fetchVmsCDRom({ vms: internalVms })
       yield fetchVmsNics({ vms: internalVms })
     } else {
-      logDebug('fetchVmsByCount() shallow fetch requested - skipping other resources')
+      logDebug('fetchVmsByCountVLower() shallow fetch requested - skipping other resources')
     }
   }
 
   yield put(persistState())
+  return fetchedVmIds
+}
+
+export function* fetchSingleVm (action) {
+  const { vmId, shallowFetch } = action.payload
+  yield startProgress({ vmId, name: 'refresh_single' })
+
+  const isOvirtGTE42 = compareVersionToCurrent({ major: 4, minor: 2 })
+  if (isOvirtGTE42 && !shallowFetch) {
+    action.payload.additional =
+      ['cdroms', 'sessions', 'disk_attachments.disk', 'graphics_consoles', 'nics']
+  }
+
+  yield put(addVmsFetchedById({ vmIds: [ vmId ] }))
```

Comment:
Do I understand correctly, that this state/actions around `*FetchedById` are just because the regular refresh reads recently the paginated data only? Means the VM requested when enetering the application via URL with VM ID would not be regularly refreshed?

If so, I would like to avoid this extra level of state information.
I might be enough for the `refreshData` to check the state for list of recent VMs (those already downloaded for whatever reason) and compute difference between those already stored and reloaded with pagination.

I believe this would be much simpler and less error prone in the future since we will maintain just a single list.

Reply:
> Do I understand correctly, that this state/actions around `*FetchedById` are just because the regular refresh reads recently the paginated data only? Means the VM requested when enetering the application via URL with VM ID would not be regularly refreshed

No only would the VM not have been refreshed by `refreshData`, but it would have been removed completely from the list.  Refresh generally follows this path:
 - Fetch the pages the user has seen
 - Merge the fetched data to the store
 - Remove VMs from the store that were not just fetched

> If so, I would like to avoid this extra level of state information.
It might be enough for the refreshData to check the state for list of recent VMs (those already downloaded for whatever reason) and compute difference between those already stored and reloaded with pagination.

So, fetch by page, subtract the page fetched VMs from the list of VMs in the store and then fetch each of those VMs individually and either update (200 response) or remove from the store (404 response)?

Sounds like that would work.  I'll try it out.